### ISSUE TITLE
Proposed updates to OSCD dataset

### ISSUE THREAD

**Issue by robmarkcole** [microsoft/torchgeo#1683]
### Summary

- Proposal 1: Following https://github.com/microsoft/torchgeo/issues/1634 the OSCD dataset should be updated to support `bands=["B04", "B03", "B02"]` rather than the current `bands in ["rgb", "all"]`

- Proposal 2: OSCD is a change dataset and currently the before/after images are concatenated. However [as discussed here](https://github.com/microsoft/torchgeo/discussions/1535) it is preferable to return these separately. Required updates to `__getitem__` below:

```python
        image1 = self._load_image(files["images1"])
        image2 = self._load_image(files["images2"])
        mask = self._load_target(str(files["mask"]))

        if self.transforms is not None:
            image = torch.cat([image1, image2])
            sample = {"image": image, "mask": mask}
            sample = self.transforms(sample)
            idx = sample["image"].shape[0] // 2
            image1 = get_masked(sample["image"][:idx])
            image2 = get_masked(sample["image"][idx:])

        sample = {"image1": image1, "image2": image2, "mask": mask}
        return sample

### Rationale

Harmonise the use of `bands` arg and prepare for a ChangeDetectionTask

### Implementation

_No response_

### Alternatives

_No response_

### Additional information

_No response_

**Comment by adamjstewart** [comment]
Agree with both of these changes. I think in #289 we decided to standardize on bands being a list of strings of band names, so any datasets that don't follow this are outdated.

**Comment by robmarkcole** [comment]
Possibly we should also consider mirroring this dataset to huggingface as the native download is sloooow 

<img width="999" alt="image" src="https://github.com/microsoft/torchgeo/assets/11855322/63939847-f28b-44a3-b0dd-a3fb2731b2d5">

**Comment by adamjstewart** [comment]
You'll need to figure out what license the data is distributed under and ensure that it allows for redistribution.
